homeassistant/components/sensoterra/config_flow.py

```diff
@@ -0,0 +1,79 @@
+"""Config flow for Sensoterra integration."""
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+from typing import Any
+
+from sensoterra.customerapi import (
+    CustomerApi,
+    InvalidAuth as StInvalidAuth,
+    Timeout as StTimeout,
+)
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_USER, ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, CONF_TOKEN
+from homeassistant.helpers.selector import (
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import DOMAIN, LOGGER, TOKEN_EXPIRATION_DAYS
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EMAIL): TextSelector(
+            TextSelectorConfig(type=TextSelectorType.EMAIL, autocomplete="email")
+        ),
+        vol.Required(CONF_PASSWORD): TextSelector(
+            TextSelectorConfig(type=TextSelectorType.PASSWORD)
+        ),
+    }
+)
+
+
+class SensoterraConfigFlow(ConfigFlow, domain=DOMAIN):
```

Comment:
Do we have something unique that we can use as unique identifier for the config flow to avoid the user to set up twice

Reply:
No, I don't think so. The API token gives access to possibly zero or more devices. The API token is not stable, nor is the email address of the customer.

Reply:
What kind of token do we get? JWT?

Reply:
Otherwise we have to do `self._async_abort_entries_match` to make sure we can't add duplicates

Reply:
As I also maintain the Sensoterra API, I have change the API token from a hash to a JWT token. This token contains the unique ID we need. I've implemented `await self.async_set_unique_id(device_unique_id)`.